test/support/sugar_client.coffee

```diff
@@ -7,42 +7,63 @@ class SugarClient
     primus = sugar.primus
     client = primus.Socket connection
     binding = { sugar, primus, client }
-    
+
     methods = [
-      'spark', 'ping', 'subscribeTo', 'unsubscribeFrom',
+      'spark', 'pong', 'subscribeTo', 'unsubscribeFrom',
       'sendEvent', 'keepAliveTimer', 'hasResponse'
     ]
-    
+
     client[method] = @[method].bind binding for method in methods
     @listenTo client
     return client
-  
+
   listenTo: (client) ->
     client.on 'data', client.hasResponse
-  
+
   spark: ->
     deferred = Bluebird.defer()
     @client.id (id) => deferred.resolve @primus.spark id
     deferred.promise
-  
-  ping: ->
+
+  pong: ->
     deferred = Bluebird.defer()
-    @client.write "primus::ping::#{ +new Date() }"
-    @client.once 'incoming::pong', -> deferred.resolve()
+    # original behaviour
+    # client would write a ping message
+    # @client.write "primus::ping::#{ +new Date() }"
+    # server would respond with a pong
+    # and client would dispatch the incoming pong event
+    # @client.once 'incoming::pong', -> deferred.resolve()
+
+    # new server -> client behaviour
+    # server sends a ping
+    # client dispatches the incoming ping event
+    @client.emit 'incoming::ping', -> deferred.resolve()
```

Comment:
I’m pretty sure this should be a listener, not an emitter.

Reply:
`@client.once` might fix the tests.

Reply:
So, my dumb solution to this is to wait for 100ms then resolve the promise. See #106.
```suggestion
    setTimeout ->
      deferred.resolve()
    , 100
```